wallet-api/src/Ledger/Tx.hs

```diff
@@ -162,8 +126,10 @@ data Tx = Tx {
     -- ^ The 'Value' forged by this transaction.
     txFee        :: !Ada,
     -- ^ The fee for this transaction.
-    txValidRange :: !SlotRange
+    txValidRange :: !SlotRange,
     -- ^ The 'SlotRange' during which this transaction may be validated.
+    txSignatures :: Map PubKey Signature
```

Comment:
I think we hash the transaction by using the derived `Serialise` instance. But we should be signing something that doesn't include the signatures, which that does. In particular, I'm not sure how we ever validate any signatures, since I'd expect us to recalculate the hash later and get a different thing...

Also, we don't want to just rely on the map being empty, since that way we can add more signatures later without having to contort ourselves.

Reply:
The hash is of a `TxStripped` which excludes the signatures.

Reply:
https://github.com/input-output-hk/plutus/blob/7ab2473b9b1856d0d8307916a85774810562a48a/wallet-api/src/Ledger/Tx.hs#L194

Although it might be better to just add a type parameter for the signatures to `Tx`...

Reply:
Or have `TxBody` and `Tx`, which is the terminology that the specifications use, I think?